### ISSUE TITLE
Logging OTLP support

### ISSUE THREAD

**Issue by a-thaler** [kyma-project/telemetry-manager#556]
**Motivation**
The telemetry module was initially designed to be based fully on the OpenTelemetry project. As the logs domain of the project was not stable yet and also the adoption is simply not there yet, the module was released having the application logs based on Fluentbit using the proprietary HTTP output. The SAP Cloud Logging backend was not supporting OTLP as well.

These criterias changed and the adoption slowly kicks-in:
- On the backends:
  - SAP Cloud Logging supports OTLP based ingestion with much better performance and much better correlation capabilities with traces and metrics.
  - Most 3party service ofeerings are supporting it like dynatrace, signalFX or honeycomb
  - More and more backends are supporting it, also Loki is in an active implementation for adding support.
- On the clients:
  - The otel-sdk is getting more and more used for instrumentation of traces and metrics and supports logs as well, so people will start trying it out.
  - Istio access logging is supporting OTLP, avoiding the mix up of the access logs with the envoy application logs via stdout.
  - SAP CAP starts using the otel-sdk
 
The client adoption brings the benefit of streamlining the setup by introducing a gateway like done for traces and metrics already, allowing clients to directly push logs without the indirection via stdout, enabling more opportunities like enabling easy collection of kubernetes event logs,

Making logs based on OTLP will allow a streamlined approach to telemetry data across traces and metrics. All data can have  streamlined attributes following streamlined semantics.

Also the proprietary protocol used with fluentbit is strongly aligned with the SAP Cloud Logging API and is not common for ingestion to other systems, so the usage is very limited.

Fluentbit as technology is different to the otel-collector framework and not that flexible. A lot of synergies can be seen in the telemetry-manager between the traces and metrics domain and could be applied to logging as well, simplifying the code base and maintenance.

**Goal and requirements**

Goal is to make logs based on the OTLP protocol (for backend ingestion to support more providers, for client ingestion to avoid indirection via stdout if not desired), leveraging the otel-collector framework to have a streamlined technology stack, still supporting collection of logs via stdout.

The requirements can be split into three parts:
- backend OTLP ingestion support
  - support of OTLP outputs similar to traces and metrics
  - data will get enriched in OTLP fashion following same enrichment logic as already done for traces and metrics, covering mainly the cluster name, the k8s resource attributes and the service name
- tail based log collection enabled by dedicated input
  - similar to metrics input, have a way to enable log collection via runtime log tailing
  - support namespace filtering
  - try a meaningful mapping of log entries to the OTLP protocol in case of JSON logs
  - optional support dedicated mapping instructions per workload
- client OTLP support (not needed for feature parity)
  - introduce an endpoint to push logs in OTLP protocol similar to metrics/traces

The new API must be available in parallel to the old API which will get deprecated but will stay till there is no usage anymore.
With https://github.com/kyma-project/kyma/issues/15932 fluentbit introduced support for OTLP and a first iteration might introduce the OTLP output based on fluenbit. However, first investigations revealed that fluentbit does neglect OTLP resource attributes, especially the enrichment of relevant k8s metadata will be not possible. With that a direct jump to the otel-collector seems to be needed (which is desired anyway).

**Targetted Architecture**
![logs drawio](https://github.com/kyma-project/telemetry-manager/assets/7479371/c29c3776-fc8a-4952-9d3f-3c8cea13da16)

**Actions**

- [x] Concept phase (not available at all)
  - [x] Build up knowledge of the OTLP log semantics and tooling and come up with a mapping concept when transforming JSON logs to OTLP (https://github.com/kyma-project/telemetry-manager/issues/691)
  - [x] Prove the derived mapping concept (https://github.com/kyma-project/telemetry-manager/issues/720)
  - [x] Prove that shipping tailed logs via the gateway can be reliable enough, performant enough, scalable enough (https://github.com/kyma-project/telemetry-manager/issues/769)
  - [x] Understand and define access log integration based on Istio OTLP support (https://github.com/kyma-project/telemetry-manager/issues/821)
- [x] Implementation phase (available in experimental release)
  - [x] Add a feature flag and have a modified LogPipeline CRD which gets active on experimental channel (https://github.com/kyma-project/telemetry-manager/issues/1415)
  - [x] Identify throughput limits for log gateway (https://github.com/kyma-project/telemetry-manager/issues/2080)
  - [x] Add k8s label support (and maybe deprecate annotation support) (https://github.com/kyma-project/telemetry-manager/issues/1802)
  - [x] Implement status sub-resource for log-gateway and log-agent. https://github.com/kyma-project/telemetry-manager/issues/2110
  - [x] New Log Gateway
       - [x] Do performance tests on the future gateway setup to understand if auto-scaling will be a mandatory requirement to run bigger cluster setups (https://github.com/kyma-project/telemetry-manager/issues/1289)
       - [x] Add a new OTLP output which reveals into a log gateway deployment (https://github.com/kyma-project/telemetry-manager/issues/872)
       - [x] Add self-monitoring capabilities for gateway (https://github.com/kyma-project/telemetry-manager/issues/1800)
       - [x] Add serviceName enrichment logic (https://github.com/kyma-project/telemetry-manager/issues/1950)
       - [x] Enrich observedTime if not present https://github.com/kyma-project/telemetry-manager/pull/1928
       - [x] Decide on new gateway considerations (https://github.com/kyma-project/telemetry-manager/pull/1855)
       - [x] OTLP input namespace filter https://github.com/kyma-project/telemetry-manager/issues/2002
  - [x] New Log agent
    - [x] Come up with a concrete log agent config and have another perf test (https://github.com/kyma-project/telemetry-manager/issues/1631)
    - [X] Add log agent support (https://github.com/kyma-project/telemetry-manager/issues/1683)
    - [x] Assure that namespace filtering works proper in a multi-pipeline setup
    - [x] Disabled agent feature will remove agent (https://github.com/kyma-project/telemetry-manager/issues/336)
    - [x] Decouple agent from gateway (https://github.com/kyma-project/telemetry-manager/issues/1854)
    - [x] Add self-monitor support for agent https://github.com/kyma-project/telemetry-manager/issues/1887
    - [x] traceID parsing https://github.com/kyma-project/telemetry-manager/issues/2004
    - [x] move body into "log.original" (instead of copying into "original" attribute) https://github.com/kyma-project/telemetry-manager/issues/2003
    - [x] Restart bug https://github.com/kyma-project/telemetry-manager/issues/2135
  - [x] Istio access logs via OTLP and related noise filter (https://github.com/kyma-project/telemetry-manager/issues/1899)
    - [x] log enrichement (https://github.com/kyma-project/telemetry-manager/issues/1977)
    - [x] noise filter (https://github.com/kyma-project/telemetry-manager/issues/1978)
    - [x] Support in Istio: https://github.com/kyma-project/istio/pull/1374
- [x] Finalization phase (available in regular release)
  - [x] Verify that in OTLP output mode the fluenbit specific API is not usable (custom filter, keepLabels, dropAnnotations) (https://github.com/kyma-project/telemetry-manager/pull/2115)
  - [x] Global enrichement block in telemetry resource (https://github.com/kyma-project/telemetry-manager/issues/2109)
  - [X] Add new end-user documentation (https://github.com/kyma-project/telemetry-manager/pull/1990)
  - [x] Add troubleshooting guides and link them in self-monitor messages
  - [x] Release-readiness (https://github.com/kyma-project/telemetry-manager/issues/2191)
- [x] Follow-Up tickets created
  - [ ] New input for k8s events (https://github.com/kyma-project/telemetry-manager/issues/560)
  - [ ] Add parser and multiline parser support (https://github.com/kyma-project/telemetry-manager/issues/1890)
  - [ ] Deprecate fluentbit API (https://github.com/kyma-project/telemetry-manager/issues/2232)
  - [ ] Have reproducible performance tests (https://github.com/kyma-project/telemetry-manager/issues/1774)
